extensions/amp-messaging-protocol/0.1/amp-messaging-protocol.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isExperimentOn} from '../../../src/experiments';
+import {user} from '../../../src/log';
+
+const TAG = 'amp-messaging-protocol';
+
+
+export class AmpMessagingProtocol extends AMP.BaseElement {
```

Comment:
How about `AmpViewerIntegration`?

I think it is important to be concrete, because we actually do a lot of messaging with child iframes as well.